add mdbootstrap@4.1.1 w/ npm auto-update

kennynaoh opened the pull request:
PR for #8906 
@x09326 please help me review this pull request, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/mdbootstrap/bootstrap-material-design
 * Official website (optional, not the repository): https://mdbootstrap.com/material-design-for-bootstrap/
 * NPM package url (optional): https://www.npmjs.com/package/mdbootstrap
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 96
   - Count of watchers: 13
   - Count of forks: 24
 * NPM download stats (optional):
   - Downloads in the last day: 22
   - Downloads in the last week: 124
   - Downloads in the last month: 124

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #8906, cc @mattonit

x09326 commented:
@kennynaoh 
I'm not sure whether the files should be added.
Do you confirm that with the author?

mattonit commented:
@x09326 
files that should be added:
css/mdb.css
css/mdb.min.css
js/mdb.js
js/mdb.min.js

PeterDaveHello commented:
@mattonit Thanks a lot!

PeterDaveHello commented:
LGTM

kennynaoh commented:
LGTM

x09326 commented:
LGTM

PeterDaveHello commented:
LGTM

kennynaoh commented:
Thanks @mattonit ,
I have added libs you mentioned.
And I have also added `font/roboto/*`, `img/overlays/*`, right?
Please help me confirm that.
Thanks!

mattonit commented:
@kennynaoh, yes `font/roboto/*` and `img/overlays/*` should be added too

thanks!

kennynaoh commented:
@x09326 
So Please help me take a look again.
Thanks!

x09326 commented:
@kennynaoh 
Please rebase it.

kennynaoh commented:
@x09326 
I have rebased to the latest master branch.
Please help me review it again.
Thanks!

x09326 commented:
LGTM

PeterDaveHello commented on ajax/libs/mdbootstrap/package.json:
```diff
@@ -0,0 +1,35 @@
+    {
+      "basePath": "",
+      "files": [
+        "+(js|css)/mdb*",
```
What about `"+(js|css)/**/mdb*"`?

kennynaoh replied:
Thanks @PeterDaveHello ,
I want to ask something.
How did you determine that we need `**` under `css|js` folder in this case?
Thanks!

PeterDaveHello replied:
We don't "need" it, it's for better compatibility, for the future relases.

PeterDaveHello commented on ajax/libs/mdbootstrap/package.json:
```diff
@@ -0,0 +1,35 @@
+      "basePath": "",
+      "files": [
+        "+(js|css)/mdb*",
+        "font/roboto/*.+(eot|ttf|woff|woff2)",
```
I prefer to replace `roboto` with `**`

kennynaoh commented:
@x09326 
I have updated the `npmFileMap`.
Please help me review it again.
Thanks!

PeterDaveHello commented:
Non-trival auto-update also needs multi reviewers.

pvnr0082t commented:
LGTM

PeterDaveHello commented:
LGTM
